Maskbook/Repository/PersonaRepository.swift

```diff
@@ -241,5 +242,153 @@ enum PersonaRepository {
             try? viewContext.saveOrRollback()
         }
     }
+ 
+    static func getPersonaCount() async -> Int {
+        await withCheckedContinuation { continuation in
+            let fetchRequest = PersonaRecord.sortedFetchRequest
+            fetchRequest.predicate = PersonaRecord.predicate(initialized: true)
+            
+            var count = 0
+            backgroundContext.performAndWait {
+                count = (try? backgroundContext.count(for: fetchRequest)) ?? 0
+            }
+            continuation.resume(returning: count)
+        }
+    }
+    
+    static func getAssociatedAccountsCount() async -> Int {
+        await withCheckedContinuation { continuation in
+            let fetchRequest = PersonaRecord.sortedFetchRequest
+            fetchRequest.predicate = PersonaRecord.hasLinkedProfiles()
+            fetchRequest.relationshipKeyPathsForPrefetching = ["linkedProfiles"]
+            
+            var count = 0
+            backgroundContext.performAndWait {
+                if let personasWithLinkedProfiels = try? backgroundContext.fetch(fetchRequest) {
+                    count = personasWithLinkedProfiels.reduce(0) { partialResult, record in
+                        partialResult + (record.linkedProfiles?.count ?? 0)
+                    }
+                }
+            }
+            continuation.resume(returning: count)
+        }
+    }
     
+    static func getPersonasForBackup() async -> [JSON] {
+        await withCheckedContinuation { continuation in
+            var personas: [JSON] = []
+            let fetchRequest = PersonaRecord.sortedFetchRequest
+            fetchRequest.predicate = PersonaRecord.predicate(
+                identifier: nil,
+                hasPrivateKey: nil,
+                hasLogout: nil,
+                nameContains: nil,
+                initialized: true)
+            
+            backgroundContext.performAndWait {
+                if let personaRecords = try? backgroundContext.fetch(fetchRequest) {
+                    personas = personaRecords.compactMap { $0.getBackupJson() }
+                }
+            }
+            continuation.resume(returning: personas)
+        }
+    }
+    
+    static func restoreFromJson(_ json: JSON) throws {
+        var restoreError: Error?
```

Comment:
`Persona` has a two functions that decode from json and create `PersonaRecord` from it. So it's a bridge of json and `PersonaRecord`.

Why not extend `Persona`'s decode method and decode manually ?

Reply:
Two reasons:
1. Existing Persona's decode function is used to decode/encode JSON from JS `public-api`, while JSON from the backup files **has a different structure**.
2. No need to do 2 times convertion (`PersonaRecord` -> `Persona` -> Backup JSON), thus I add a new function to generate to/restore from JSON